### PR TITLE
Image downloader Dockerfile to meet CPaaS requirements

### PR DESCRIPTION
**Pull request by mpryc**
Updated Dockerfile for the image_downloader container that
allows to translate from upstream to downstream using
dockerfile_to_osbs.sh from openstack-k8s-operators-ci.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -52,7 +45,7 @@ LABEL com.redhat.component="osp-director-operator-container" \
 
 ENV USER_UID=1001 \
     OPERATOR_TEMPLATES=/usr/share/osp-director-operator/templates/ \
-    WATCH_NAMESPACE=openstack,openshift-machine-api,openshift-sriov-network-operator
```
Comment (yazug) [inline Dockerfile:55]: is this intentional change for this effort? This change should likely be driven based on the operator change not a rework for layout/structure of the Dockerfile

Comment (mpryc) [reply Dockerfile:55]: @yazug This is what I found in the midstream repo, but I can revert back to the upstream version of it.

Comment (yazug) [reply Dockerfile:55]: use upstream version please.

File: Dockerfile.provision-ip-discovery-agent [hunk 2]
```diff
@@ -1,15 +1,45 @@
-FROM registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.7 AS builder
-WORKDIR /go/src/github.com/openstack-k8s-operators/osp-director-operator
-COPY . .
-RUN go mod vendor
-RUN BIN_PATH=build/_output/cmd ./containers/provision_ip_discovery_agent/build-provision-ip-discovery-agent.sh
```
Comment (yazug) [inline Dockerfile.provision-ip-discovery-agent:5]: hmm If I remember correctly this script does have some additional version injection stuff that wasn't done in here so not sure how to resolve the discrepancy and should probably get input from maintainer of that build script

File: Dockerfile.provision-ip-discovery-agent [hunk 3]
```diff
@@ -1,15 +1,45 @@
...
+LABEL com.redhat.component="osp-director-provisioner-container" \
+      name="osp-director-provisioner-container" \
+      version="1.0"
+
+WORKDIR /
+
+# Install binary to WORKDIR
+COPY --from=builder ${DEST_ROOT}/${WHAT} .
+
+ENTRYPOINT ["/provision-ip-discovery-agent"]
```
Comment (yazug) [inline Dockerfile.provision-ip-discovery-agent:45]: hmm we flipped from CMD to ENTRYPOINT not sure if this is workable flip or improvement?

File: Dockerfile.provision-ip-discovery-agent (hunk 2 above)
Comment (mpryc) [reply Dockerfile.provision-ip-discovery-agent:5]: I have used CPaaS version of Dockerfile rather then logic from here, I can add it, will check with @abays if it's required.

File: Dockerfile.provision-ip-discovery-agent (hunk 3 above)
Comment (mpryc) [reply Dockerfile.provision-ip-discovery-agent:45]: It's almost the same, used ENTRYPOINT to be similar to operator Dockerfile as those two images should not be much different.

File: Dockerfile.provision-ip-discovery-agent (hunk 2 above)
Comment (abays) [reply Dockerfile.provision-ip-discovery-agent:5]: The `VERSION_OVERRIDE` in the build script `containers/provision_ip_discovery_agent/build-provision-ip-discovery-agent.sh` did not end up ever being used, if I recall correctly.  Is this the version injection to which you were referring, @yazug?

Comment (yazug) [reply Dockerfile.provision-ip-discovery-agent:5]: yes, that is the part I was referring to, can we prune that out or the script out if this is sufficient?

Comment (abays) [reply Dockerfile.provision-ip-discovery-agent:5]: Yeah, just ignore that part of the bash script.  I borrowed it from somewhere else where the versioning stuff must have been needed, but then I didn't bother to pare it down for our version.

**Comment by abays** [comment]
The two containers build and function as expected in my environment

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**